contents/docs/self-host/configure/migrate-postgres-clickhouse.mdx

```diff
@@ -0,0 +1,82 @@
+---
+title: Migrate from Postgres to ClickHouse
+sidebar: Docs
+showTitle: true
+---
+
+> This documentation is in **Beta**. Please provide feedback via the [PostHog Communty Slack workspace](/slack) or [via a GitHub issue](https://github.com/PostHog/posthog.com/issues).
+
+**PostHog backed by Postgres is now deprecated**. We will continue to provide support to Postgres backed installs but we strongly encourage you to migrate to PostHog backed by ClickHouse for improved performance, to receive new features, and continued support in the future.
+
+## Before you begin
+
+TODOs:
+- Is there a minimum version for PostHog + Postgres and a minimum version for PostHog + ClickHouse? For example, does the Replicator plugin have a minimum PostHog version to work within?
+
+In order to migrate you must have the following in place:
+
+1. The PostHog + Postgres installation
+2. A new PostHog + ClickHouse installation
+
+## Export
+
+TODOs:
+- Are these the correct tables to dump?
+- Should we be using `--data-only` to only dump the data and not the schema creation?
+
+The first step in the migration process is to create a Postgres database dump. With a PostHog + Postgres instance up and running, run the `pg_dump` command with the following format:
+
+```
+$ pg_dump -d [database] -h [host] -U [user] -T posthog_event -T posthog_person -T posthog_sessionrecordingevent -f [output-file]
+```
+
+For example:
+
+```
+$ pg_dump -d posthog -h localhost -U posthog -T posthog_event -T posthog_person -T posthog_sessionrecordingevent -f posthog-migration.sql
+```
+
+The example command above creates a `posthog-migration.sql` that contains SQL statements to create the schema and insert the data data, excluding event data, from the existing Postgres database.
+
+## Import
+
+The next step is to get the data into the new PostHog + ClickHouse installation. This installation still utilizes Postgres, but only for non-event data. Event data is now stored in ClickHouse.
+
+The import process has two steps:
+
+1. Run the exported SQL against the Postgres database for the new installation
+2. Use the PostHog replicator plugin to migrate event data from the old PostHog instance to the new one
+
+### Postgres import of non-event data
+
+Import the old Postgres data into the new PostHog Postgres database. The command should take the following format:
+
+```
+pg_restore -d [database] -h [host] -p [port] -U [user] [input-file]
+```
+
+For example:
+
+```
+pg_restore -d posthog -h remote.example.com -p 5432 -U posthog posthog-migration.sql
```

Comment:
As is, the output of this is as follows. I think we're going to need someone who has a strong understanding of our DB schema to help out there. @timgl can you help or recommend someone else to?

```
Unable to use a TTY - input is not a terminal or the right kind of file
SET
SET
SET
SET
SET
 set_config
------------

(1 row)

SET
SET
SET
SET
CREATE EXTENSION
ERROR:  schema "graphile_worker" already exists
ERROR:  role "posthog" does not exist
COMMENT
CREATE EXTENSION
COMMENT
SET
ERROR:  unrecognized configuration parameter "default_table_access_method"
ERROR:  relation "jobs" already exists
ERROR:  role "posthog" does not exist
ERROR:  function "add_job" already exists with same argument types
ERROR:  role "posthog" does not exist
ERROR:  function "complete_job" already exists with same argument types
ERROR:  role "posthog" does not exist
ERROR:  function "complete_jobs" already exists with same argument types
ERROR:  role "posthog" does not exist
ERROR:  function "fail_job" already exists with same argument types
ERROR:  role "posthog" does not exist
ERROR:  function "get_job" already exists with same argument types
ERROR:  role "posthog" does not exist
ERROR:  function "jobs__decrease_job_queue_count" already exists with same argument types
ERROR:  role "posthog" does not exist
ERROR:  function "jobs__increase_job_queue_count" already exists with same argument types
ERROR:  role "posthog" does not exist
ERROR:  function "permanently_fail_jobs" already exists with same argument types
ERROR:  role "posthog" does not exist
ERROR:  function "remove_job" already exists with same argument types
ERROR:  role "posthog" does not exist
ERROR:  function "reschedule_jobs" already exists with same argument types
ERROR:  role "posthog" does not exist
ERROR:  function "tg__update_timestamp" already exists with same argument types
ERROR:  role "posthog" does not exist
ERROR:  function "tg_jobs__notify_new_jobs" already exists with same argument types
ERROR:  role "posthog" does not exist
ERROR:  relation "job_queues" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "jobs_id_seq" already exists
ERROR:  role "posthog" does not exist
ALTER SEQUENCE
ERROR:  relation "known_crontabs" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "migrations" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "auth_group" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "auth_group_id_seq" already exists
ERROR:  role "posthog" does not exist
ALTER SEQUENCE
ERROR:  relation "auth_group_permissions" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "auth_group_permissions_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "auth_permission" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "auth_permission_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "axes_accessattempt" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "axes_accessattempt_id_seq" already exists
ALTER SEQUENCE
ERROR:  role "posthog" does not exist
ALTER SEQUENCE
ALTER SEQUENCE
ERROR:  relation "axes_accesslog" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "axes_accesslog_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "django_admin_log" already exists
ALTER SEQUENCE
ERROR:  role "posthog" does not exist
ALTER SEQUENCE
ERROR:  relation "django_admin_log_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "django_content_type" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "django_content_type_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "django_migrations" already exists
ERROR:  role "posthog" does not exist
ALTER SEQUENCE
ALTER SEQUENCE
ERROR:  relation "django_migrations_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "django_session" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "ee_enterpriseeventdefinition" already exists
ALTER SEQUENCE
ERROR:  role "posthog" does not exist
ERROR:  relation "ee_enterprisepropertydefinition" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "ee_hook" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "ee_license" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "ee_license_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_action" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_action_events" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_action_events_id_seq" already exists
ERROR:  role "posthog" does not exist
ALTER SEQUENCE
ALTER SEQUENCE
ERROR:  relation "posthog_action_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_actionstep" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_actionstep_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_annotation" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_annotation_id_seq" already exists
ERROR:  role "posthog" does not exist
ALTER SEQUENCE
ERROR:  relation "posthog_cohort" already exists
ERROR:  role "posthog" does not exist
ALTER SEQUENCE
ALTER SEQUENCE
ERROR:  relation "posthog_cohort_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_cohortpeople" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_cohortpeople_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_dashboard" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_dashboard_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_dashboarditem" already exists
ERROR:  role "posthog" does not exist
ALTER SEQUENCE
ALTER SEQUENCE
ERROR:  relation "posthog_dashboarditem_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_element" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_element_id_seq" already exists
ALTER SEQUENCE
ALTER SEQUENCE
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_elementgroup" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_elementgroup_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_event_id_seq" already exists
ERROR:  role "posthog" does not exist
ALTER SEQUENCE
ALTER SEQUENCE
ERROR:  relation "posthog_eventdefinition" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_featureflag" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_featureflag_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_messagingrecord" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_organization" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_organizationinvite" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_organizationmembership" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_person_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_personalapikey" already exists
ALTER SEQUENCE
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_persondistinctid" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_persondistinctid_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_plugin" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_plugin_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_pluginattachment" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_pluginattachment_id_seq" already exists
ALTER SEQUENCE
ALTER SEQUENCE
ALTER SEQUENCE
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_pluginconfig" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_pluginconfig_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_pluginlogentry" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_pluginstorage" already exists
ERROR:  role "posthog" does not exist
ALTER SEQUENCE
ERROR:  relation "posthog_pluginstorage_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_propertydefinition" already exists
ALTER SEQUENCE
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_sessionrecordingevent_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_sessionrecordingviewed" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_sessionrecordingviewed_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_sessionsfilter" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_sessionsfilter_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_team" already exists
ALTER SEQUENCE
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_team_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_team_users" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_team_users_id_seq" already exists
ALTER SEQUENCE
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_user" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_user_groups" already exists
ERROR:  role "posthog" does not exist
ALTER SEQUENCE
ERROR:  relation "posthog_user_groups_id_seq" already exists
ALTER SEQUENCE
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_user_id_seq" already exists
ERROR:  role "posthog" does not exist
ALTER SEQUENCE
ERROR:  relation "posthog_user_user_permissions" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "posthog_user_user_permissions_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "rest_hooks_hook" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "rest_hooks_hook_id_seq" already exists
ALTER SEQUENCE
ERROR:  role "posthog" does not exist
ERROR:  relation "social_auth_association" already exists
ERROR:  role "posthog" does not exist
ALTER SEQUENCE
ALTER SEQUENCE
ALTER SEQUENCE
ERROR:  relation "social_auth_association_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "social_auth_code" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "social_auth_code_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "social_auth_nonce" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "social_auth_nonce_id_seq" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "social_auth_partial" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "social_auth_partial_id_seq" already exists
ALTER SEQUENCE
ALTER SEQUENCE
ERROR:  role "posthog" does not exist
ERROR:  relation "social_auth_usersocialauth" already exists
ERROR:  role "posthog" does not exist
ERROR:  relation "social_auth_usersocialauth_id_seq" already exists
ERROR:  role "posthog" does not exist
ALTER SEQUENCE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
ALTER TABLE
COPY 0
COPY 0
COPY 0
ERROR:  duplicate key value violates unique constraint "migrations_pkey"
DETAIL:  Key (id)=(1) already exists.
CONTEXT:  COPY migrations, line 1
COPY 0
COPY 0
ERROR:  duplicate key value violates unique constraint "auth_permission_pkey"
DETAIL:  Key (id)=(1) already exists.
CONTEXT:  COPY auth_permission, line 1
COPY 0
COPY 0
COPY 0
ERROR:  duplicate key value violates unique constraint "django_content_type_pkey"
DETAIL:  Key (id)=(1) already exists.
CONTEXT:  COPY django_content_type, line 1
ERROR:  duplicate key value violates unique constraint "django_migrations_pkey"
DETAIL:  Key (id)=(1) already exists.
CONTEXT:  COPY django_migrations, line 1
COPY 1
COPY 0
COPY 0
COPY 0
COPY 0
ERROR:  insert or update on table "posthog_action" violates foreign key constraint "posthog_action_team_id_3a21e3a6_fk_posthog_team_id"
DETAIL:  Key (team_id)=(2) is not present in table "posthog_team".
ERROR:  insert or update on table "posthog_action_events" violates foreign key constraint "posthog_action_events_action_id_f6f1f077_fk_posthog_action_id"
DETAIL:  Key (action_id)=(8) is not present in table "posthog_action".
ERROR:  insert or update on table "posthog_actionstep" violates foreign key constraint "posthog_actionstep_action_id_b50d75e6_fk_posthog_action_id"
DETAIL:  Key (action_id)=(1) is not present in table "posthog_action".
COPY 0
COPY 0
COPY 0
ERROR:  insert or update on table "posthog_dashboard" violates foreign key constraint "posthog_dashboard_team_id_5c28ee1a_fk_posthog_team_id"
DETAIL:  Key (team_id)=(2) is not present in table "posthog_team".
ERROR:  null value in column "favorited" violates not-null constraint
DETAIL:  Failing row contains (1, Daily Active Users (DAUs), {"events": [{"id": "$pageview", "math": "dau", "name": "$pagevie..., null, null, f, 1, 1, {}, null, 2021-07-29 14:14:35.969525+00, f, null, 2021-07-29 14:14:35.970309+00, null, f, Shows the number of unique users that use your app everyday., f, cache_d8975ac4e6290c3ff0e8044f9f733983, F-zo4q-V, null, null, null, null).
CONTEXT:  COPY posthog_dashboarditem, line 1: "1	Daily Active Users (DAUs)	{"events": [{"id": "$pageview", "math": "dau", "name": "$pageview", "typ..."
ERROR:  insert or update on table "posthog_element" violates foreign key constraint "posthog_element_group_id_09134876_fk_posthog_elementgroup_id"
DETAIL:  Key (group_id)=(1) is not present in table "posthog_elementgroup".
COPY 3
COPY 0
COPY 0
ERROR:  insert or update on table "posthog_eventdefinition" violates foreign key constraint "posthog_eventdefinition_team_id_818ed0f2_fk_posthog_team_id"
DETAIL:  Key (team_id)=(2) is not present in table "posthog_team".
COPY 0
ERROR:  null value in column "is_member_join_email_enabled" violates not-null constraint
DETAIL:  Failing row contains (017af29d-e39f-0000-8383-7a731cadcc6c, PostHog, 2021-07-29 14:14:31.598977+00, 2021-09-17 14:10:39.096939+00, {}, t, 9, f, {}, {}, null).
CONTEXT:  COPY posthog_organization, line 1: "017af29d-e39f-0000-8383-7a731cadcc6c	PostHog	2021-07-29 14:14:31.598977+00	2021-09-17 14:10:39.09693..."
ERROR:  insert or update on table "posthog_organizationmembership" violates foreign key constraint "posthog_organization_organization_id_6f92360d_fk_posthog_o"
DETAIL:  Key (organization_id)=(017af29d-e39f-0000-8383-7a731cadcc6c) is not present in table "posthog_organization".
COPY 0
ERROR:  duplicate key value violates unique constraint "posthog_persondistinctid_pkey"
DETAIL:  Key (id)=(1) already exists.
CONTEXT:  COPY posthog_persondistinctid, line 1
ERROR:  duplicate key value violates unique constraint "posthog_plugin_pkey"
DETAIL:  Key (id)=(1) already exists.
CONTEXT:  COPY posthog_plugin, line 1
COPY 1
ERROR:  duplicate key value violates unique constraint "posthog_pluginconfig_pkey"
DETAIL:  Key (id)=(1) already exists.
CONTEXT:  COPY posthog_pluginconfig, line 1
ERROR:  insert or update on table "posthog_pluginlogentry" violates foreign key constraint "posthog_pluginlogent_plugin_config_id_2fe30023_fk_posthog_p"
DETAIL:  Key (plugin_config_id)=(2) is not present in table "posthog_pluginconfig".
COPY 0
ERROR:  duplicate key value violates unique constraint "posthog_propertydefinition_team_id_name_e21599fc_uniq"
DETAIL:  Key (team_id, name)=(1, $plugins_deferred) already exists.
CONTEXT:  COPY posthog_propertydefinition, line 123
ERROR:  duplicate key value violates unique constraint "posthog_team_pkey"
DETAIL:  Key (id)=(1) already exists.
CONTEXT:  COPY posthog_team, line 1
COPY 0
COPY 0
COPY 0
COPY 0
COPY 0
ERROR:  insert or update on table "posthog_user" violates foreign key constraint "posthog_user_current_team_id_52760528_fk_posthog_team_id"
DETAIL:  Key (current_team_id)=(2) is not present in table "posthog_team".
COPY 0
COPY 0
COPY 0
COPY 0
COPY 0
COPY 0
 setval
--------
      1
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
    188
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
     47
(1 row)

 setval
--------
    214
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
    766
(1 row)

 setval
--------
      8
(1 row)

 setval
--------
      8
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
      4
(1 row)

 setval
--------
     11
(1 row)

 setval
--------
     26
(1 row)

 setval
--------
      3
(1 row)

 setval
--------
    919
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
    168
(1 row)

 setval
--------
    169
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
      2
(1 row)

 setval
--------
      2
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
     62
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
      2
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
      1
(1 row)

 setval
--------
      1
(1 row)

ERROR:  multiple primary keys for table "job_queues" are not allowed
ERROR:  relation "jobs_key_key" already exists
ERROR:  multiple primary keys for table "jobs" are not allowed
ERROR:  multiple primary keys for table "known_crontabs" are not allowed
ERROR:  multiple primary keys for table "migrations" are not allowed
ERROR:  relation "auth_group_name_key" already exists
ERROR:  relation "auth_group_permissions_group_id_permission_id_0cd325b0_uniq" already exists
ERROR:  multiple primary keys for table "auth_group_permissions" are not allowed
ERROR:  multiple primary keys for table "auth_group" are not allowed
ERROR:  relation "auth_permission_content_type_id_codename_01ab375a_uniq" already exists
ERROR:  multiple primary keys for table "auth_permission" are not allowed
ERROR:  multiple primary keys for table "axes_accessattempt" are not allowed
ERROR:  multiple primary keys for table "axes_accesslog" are not allowed
ERROR:  multiple primary keys for table "django_admin_log" are not allowed
ERROR:  relation "django_content_type_app_label_model_76bd3d3b_uniq" already exists
ERROR:  multiple primary keys for table "django_content_type" are not allowed
ERROR:  multiple primary keys for table "django_migrations" are not allowed
ERROR:  multiple primary keys for table "django_session" are not allowed
ERROR:  multiple primary keys for table "ee_enterpriseeventdefinition" are not allowed
ERROR:  multiple primary keys for table "ee_enterprisepropertydefinition" are not allowed
ERROR:  multiple primary keys for table "ee_hook" are not allowed
ERROR:  multiple primary keys for table "ee_license" are not allowed
ERROR:  relation "posthog_action_events_action_id_event_id_18e163e2_uniq" already exists
ERROR:  multiple primary keys for table "posthog_action_events" are not allowed
ERROR:  multiple primary keys for table "posthog_action" are not allowed
ERROR:  multiple primary keys for table "posthog_actionstep" are not allowed
ERROR:  multiple primary keys for table "posthog_annotation" are not allowed
ERROR:  multiple primary keys for table "posthog_cohort" are not allowed
ERROR:  multiple primary keys for table "posthog_cohortpeople" are not allowed
ERROR:  multiple primary keys for table "posthog_dashboard" are not allowed
ERROR:  multiple primary keys for table "posthog_dashboarditem" are not allowed
ERROR:  relation "posthog_dashboarditem_team_id_short_id_1b8c199d_uniq" already exists
ERROR:  multiple primary keys for table "posthog_element" are not allowed
ERROR:  multiple primary keys for table "posthog_elementgroup" are not allowed
ERROR:  multiple primary keys for table "posthog_eventdefinition" are not allowed
ERROR:  relation "posthog_eventdefinition_team_id_name_80fa0b87_uniq" already exists
ERROR:  multiple primary keys for table "posthog_featureflag" are not allowed
ERROR:  relation "posthog_messagingrecord_email_hash_campaign_key_6639a13f_uniq" already exists
ERROR:  multiple primary keys for table "posthog_messagingrecord" are not allowed
ERROR:  multiple primary keys for table "posthog_organization" are not allowed
ERROR:  multiple primary keys for table "posthog_organizationinvite" are not allowed
ERROR:  multiple primary keys for table "posthog_organizationmembership" are not allowed
ERROR:  multiple primary keys for table "posthog_personalapikey" are not allowed
ERROR:  relation "posthog_personalapikey_value_key" already exists
ERROR:  multiple primary keys for table "posthog_persondistinctid" are not allowed
ERROR:  multiple primary keys for table "posthog_plugin" are not allowed
ERROR:  multiple primary keys for table "posthog_pluginattachment" are not allowed
ERROR:  multiple primary keys for table "posthog_pluginconfig" are not allowed
ERROR:  multiple primary keys for table "posthog_pluginlogentry" are not allowed
ERROR:  multiple primary keys for table "posthog_pluginstorage" are not allowed
ERROR:  multiple primary keys for table "posthog_propertydefinition" are not allowed
ERROR:  relation "posthog_propertydefinition_team_id_name_e21599fc_uniq" already exists
ERROR:  relation "posthog_sessionrecording_team_id_user_id_session__ac83dfde_uniq" already exists
ERROR:  multiple primary keys for table "posthog_sessionrecordingviewed" are not allowed
ERROR:  multiple primary keys for table "posthog_sessionsfilter" are not allowed
ERROR:  relation "posthog_team_api_token_a9a1df8a_uniq" already exists
ERROR:  multiple primary keys for table "posthog_team" are not allowed
ERROR:  multiple primary keys for table "posthog_team_users" are not allowed
ERROR:  relation "posthog_team_users_team_id_user_id_c8bb8e32_uniq" already exists
ERROR:  relation "posthog_team_uuid_daa094af_uniq" already exists
ERROR:  relation "posthog_unique_plugin_storage_key" already exists
ERROR:  relation "posthog_user_distinct_id_dcd52541_uniq" already exists
ERROR:  relation "posthog_user_email_af269794_uniq" already exists
ERROR:  multiple primary keys for table "posthog_user_groups" are not allowed
ERROR:  relation "posthog_user_groups_user_id_group_id_e8f9e2be_uniq" already exists
ERROR:  multiple primary keys for table "posthog_user" are not allowed
ERROR:  relation "posthog_user_temporary_token_9d7b57f3_uniq" already exists
ERROR:  relation "posthog_user_user_permis_user_id_permission_id_1dc68b66_uniq" already exists
ERROR:  multiple primary keys for table "posthog_user_user_permissions" are not allowed
ERROR:  relation "posthog_user_uuid_cbc0c307_uniq" already exists
ERROR:  multiple primary keys for table "rest_hooks_hook" are not allowed
ERROR:  multiple primary keys for table "social_auth_association" are not allowed
ERROR:  relation "social_auth_association_server_url_handle_078befa2_uniq" already exists
ERROR:  relation "social_auth_code_email_code_801b2d02_uniq" already exists
ERROR:  multiple primary keys for table "social_auth_code" are not allowed
ERROR:  multiple primary keys for table "social_auth_nonce" are not allowed
ERROR:  relation "social_auth_nonce_server_url_timestamp_salt_f6284463_uniq" already exists
ERROR:  multiple primary keys for table "social_auth_partial" are not allowed
ERROR:  multiple primary keys for table "social_auth_usersocialauth" are not allowed
ERROR:  relation "social_auth_usersocialauth_provider_uid_e6b5e668_uniq" already exists
ERROR:  relation "unique distinct_id for team" already exists
ERROR:  relation "unique hash for each team" already exists
ERROR:  relation "unique key for team" already exists
ERROR:  relation "unique_organization_membership" already exists
ERROR:  relation "jobs_priority_run_at_id_locked_at_without_failures_idx" already exists
ERROR:  relation "auth_group_name_a6ea08ec_like" already exists
ERROR:  relation "auth_group_permissions_group_id_b120cbf9" already exists
ERROR:  relation "auth_group_permissions_permission_id_84c5c92e" already exists
ERROR:  relation "auth_permission_content_type_id_2f476e4b" already exists
ERROR:  relation "axes_accessattempt_ip_address_10922d9c" already exists
ERROR:  relation "axes_accessattempt_user_agent_ad89678b" already exists
ERROR:  relation "axes_accessattempt_user_agent_ad89678b_like" already exists
ERROR:  relation "axes_accessattempt_username_3f2d4ca0" already exists
ERROR:  relation "axes_accessattempt_username_3f2d4ca0_like" already exists
ERROR:  relation "axes_accesslog_ip_address_86b417e5" already exists
ERROR:  relation "axes_accesslog_user_agent_0e659004" already exists
ERROR:  relation "axes_accesslog_user_agent_0e659004_like" already exists
ERROR:  relation "axes_accesslog_username_df93064b" already exists
ERROR:  relation "axes_accesslog_username_df93064b_like" already exists
ERROR:  relation "django_admin_log_content_type_id_c4bce8eb" already exists
ERROR:  relation "django_admin_log_user_id_c564eba6" already exists
ERROR:  relation "django_session_expire_date_a5c62663" already exists
ERROR:  relation "django_session_session_key_c0390e0f_like" already exists
ERROR:  relation "ee_enterpriseeventdefinition_owner_id_67c5ef56" already exists
ERROR:  relation "ee_enterpriseeventdefinition_updated_by_id_7aede29e" already exists
ERROR:  relation "ee_enterprisepropertydefinition_updated_by_id_96e0be9d" already exists
ERROR:  relation "ee_hook_event_12ccd200" already exists
ERROR:  relation "ee_hook_event_12ccd200_like" already exists
ERROR:  relation "ee_hook_id_d4e48550_like" already exists
ERROR:  relation "ee_hook_team_id_638d0223" already exists
ERROR:  relation "ee_hook_user_id_442b7c5d" already exists
ERROR:  relation "index_event_definition_name" already exists
ERROR:  relation "index_property_definition_name" already exists
ERROR:  relation "only_one_owner_per_organization" already exists
ERROR:  relation "posthog_act_team_id_8c04de_idx" already exists
ERROR:  relation "posthog_action_created_by_id_4e7145d9" already exists
ERROR:  relation "posthog_action_events_action_id_f6f1f077" already exists
ERROR:  relation "posthog_action_events_event_id_7077ea70" already exists
ERROR:  relation "posthog_action_team_id_3a21e3a6" already exists
ERROR:  relation "posthog_actionstep_action_id_b50d75e6" already exists
ERROR:  relation "posthog_annotation_created_by_id_1b9c9223" already exists
ERROR:  relation "posthog_annotation_dashboard_item_id_8bfb9aa9" already exists
ERROR:  relation "posthog_annotation_organization_id_f5c9d877" already exists
ERROR:  relation "posthog_annotation_team_id_95adb263" already exists
ERROR:  relation "posthog_coh_cohort__89c25f_idx" already exists
ERROR:  relation "posthog_cohort_created_by_id_e003077a" already exists
ERROR:  relation "posthog_cohort_team_id_8a849f3d" already exists
ERROR:  relation "posthog_cohortpeople_cohort_id_1f371733" already exists
ERROR:  relation "posthog_cohortpeople_person_id_33da7d3f" already exists
ERROR:  relation "posthog_dashboard_created_by_id_a528d2ca" already exists
ERROR:  relation "posthog_dashboard_team_id_5c28ee1a" already exists
ERROR:  relation "posthog_dashboarditem_created_by_id_a3c81ce2" already exists
ERROR:  relation "posthog_dashboarditem_dashboard_id_59e2811f" already exists
ERROR:  relation "posthog_dashboarditem_team_id_a0e4bed0" already exists
ERROR:  relation "posthog_element_event_id_bb6549a0" already exists
ERROR:  relation "posthog_element_group_id_09134876" already exists
ERROR:  relation "posthog_elementgroup_team_id_3ced0286" already exists
ERROR:  relation "posthog_eventdefinition_team_id_818ed0f2" already exists
ERROR:  relation "posthog_featureflag_created_by_id_4571fe1a" already exists
ERROR:  relation "posthog_featureflag_team_id_51e383b9" already exists
ERROR:  relation "posthog_organizationinvite_created_by_id_16cbc2ef" already exists
ERROR:  relation "posthog_organizationinvite_organization_id_15e2bab4" already exists
ERROR:  relation "posthog_organizationinvite_target_email_45fa23f6" already exists
ERROR:  relation "posthog_organizationinvite_target_email_45fa23f6_like" already exists
ERROR:  relation "posthog_organizationmembership_organization_id_6f92360d" already exists
ERROR:  relation "posthog_organizationmembership_user_id_aab9f3c7" already exists
ERROR:  relation "posthog_personalapikey_id_7b00e9cc_like" already exists
ERROR:  relation "posthog_personalapikey_team_id_813f490c" already exists
ERROR:  relation "posthog_personalapikey_user_id_730a29e7" already exists
ERROR:  relation "posthog_personalapikey_value_249c258b_like" already exists
ERROR:  relation "posthog_persondistinctid_person_id_5d655bba" already exists
ERROR:  relation "posthog_persondistinctid_team_id_46330ec9" already exists
ERROR:  relation "posthog_plu_plugin__736133_idx" already exists
ERROR:  relation "posthog_plugin_organization_id_d040b9a9" already exists
ERROR:  relation "posthog_pluginattachment_plugin_config_id_cc94a1b9" already exists
ERROR:  relation "posthog_pluginattachment_team_id_415eacc7" already exists
ERROR:  relation "posthog_pluginconfig_plugin_id_d014ca1c" already exists
ERROR:  relation "posthog_pluginconfig_team_id_71185766" already exists
ERROR:  relation "posthog_pluginlogentry_message_3fb95ccc" already exists
ERROR:  relation "posthog_pluginlogentry_message_3fb95ccc_like" already exists
ERROR:  relation "posthog_pluginlogentry_plugin_config_id_2fe30023" already exists
ERROR:  relation "posthog_pluginlogentry_plugin_id_42aaf74e" already exists
ERROR:  relation "posthog_pluginlogentry_team_id_5cab32b2" already exists
ERROR:  relation "posthog_pluginstorage_plugin_config_id_6744363a" already exists
ERROR:  relation "posthog_propertydefinition_team_id_b7abe702" already exists
ERROR:  relation "posthog_ses_team_id_453d24_idx" already exists
ERROR:  relation "posthog_ses_team_id_465af1_idx" already exists
ERROR:  relation "posthog_sessionrecordingviewed_team_id_5d0d59b9" already exists
ERROR:  relation "posthog_sessionrecordingviewed_user_id_ef83047a" already exists
ERROR:  relation "posthog_sessionsfilter_created_by_id_06700ae9" already exists
ERROR:  relation "posthog_sessionsfilter_team_id_59837f38" already exists
ERROR:  relation "posthog_team_api_token_a9a1df8a_like" already exists
ERROR:  relation "posthog_team_organization_id_41bbc1d0" already exists
ERROR:  relation "posthog_team_users_team_id_2bd01272" already exists
ERROR:  relation "posthog_team_users_user_id_c7bafb47" already exists
ERROR:  relation "posthog_user_current_organization_id_e8527570" already exists
ERROR:  relation "posthog_user_current_team_id_52760528" already exists
ERROR:  relation "posthog_user_distinct_id_dcd52541_like" already exists
ERROR:  relation "posthog_user_email_af269794_like" already exists
ERROR:  relation "posthog_user_groups_group_id_75d77987" already exists
ERROR:  relation "posthog_user_groups_user_id_6b000a50" already exists
ERROR:  relation "posthog_user_temporary_token_9d7b57f3_like" already exists
ERROR:  relation "posthog_user_user_permissions_permission_id_21270994" already exists
ERROR:  relation "posthog_user_user_permissions_user_id_f94166b7" already exists
ERROR:  relation "rest_hooks_hook_event_c9f86f88" already exists
ERROR:  relation "rest_hooks_hook_event_c9f86f88_like" already exists
ERROR:  relation "rest_hooks_hook_user_id_94046fce" already exists
ERROR:  relation "single_for_internal_metrics" already exists
ERROR:  relation "social_auth_code_code_a2393167" already exists
ERROR:  relation "social_auth_code_code_a2393167_like" already exists
ERROR:  relation "social_auth_code_timestamp_176b341f" already exists
ERROR:  relation "social_auth_partial_timestamp_50f2119f" already exists
ERROR:  relation "social_auth_partial_token_3017fea3" already exists
ERROR:  relation "social_auth_partial_token_3017fea3_like" already exists
ERROR:  relation "social_auth_usersocialauth_uid_796e51dc" already exists
ERROR:  relation "social_auth_usersocialauth_uid_796e51dc_like" already exists
ERROR:  relation "social_auth_usersocialauth_user_id_17d28448" already exists
ERROR:  trigger "_100_timestamps" for relation "jobs" already exists
ERROR:  trigger "_500_decrease_job_queue_count" for relation "jobs" already exists
ERROR:  trigger "_500_decrease_job_queue_count_update" for relation "jobs" already exists
ERROR:  trigger "_500_increase_job_queue_count" for relation "jobs" already exists
ERROR:  trigger "_500_increase_job_queue_count_update" for relation "jobs" already exists
ERROR:  trigger "_900_notify_worker" for relation "jobs" already exists
ERROR:  constraint "auth_group_permissio_permission_id_84c5c92e_fk_auth_perm" for relation "auth_group_permissions" already exists
ERROR:  constraint "auth_group_permissions_group_id_b120cbf9_fk_auth_group_id" for relation "auth_group_permissions" already exists
ERROR:  constraint "auth_permission_content_type_id_2f476e4b_fk_django_co" for relation "auth_permission" already exists
ERROR:  constraint "django_admin_log_content_type_id_c4bce8eb_fk_django_co" for relation "django_admin_log" already exists
ERROR:  constraint "django_admin_log_user_id_c564eba6_fk_posthog_user_id" for relation "django_admin_log" already exists
ERROR:  constraint "ee_enterpriseeventde_eventdefinition_ptr__5b4562bd_fk_posthog_e" for relation "ee_enterpriseeventdefinition" already exists
ERROR:  constraint "ee_enterpriseeventde_owner_id_67c5ef56_fk_posthog_u" for relation "ee_enterpriseeventdefinition" already exists
ERROR:  constraint "ee_enterpriseeventde_updated_by_id_7aede29e_fk_posthog_u" for relation "ee_enterpriseeventdefinition" already exists
ERROR:  constraint "ee_enterprisepropert_propertydefinition_p_27d6057a_fk_posthog_p" for relation "ee_enterprisepropertydefinition" already exists
ERROR:  constraint "ee_enterprisepropert_updated_by_id_96e0be9d_fk_posthog_u" for relation "ee_enterprisepropertydefinition" already exists
ERROR:  constraint "ee_hook_team_id_638d0223_fk_posthog_team_id" for relation "ee_hook" already exists
ERROR:  constraint "ee_hook_user_id_442b7c5d_fk_posthog_user_id" for relation "ee_hook" already exists
ERROR:  constraint "posthog_action_created_by_id_4e7145d9_fk_posthog_user_id" for relation "posthog_action" already exists
ERROR:  constraint "posthog_action_events_action_id_f6f1f077_fk_posthog_action_id" for relation "posthog_action_events" already exists
ERROR:  constraint "posthog_action_events_event_id_7077ea70_fk_posthog_event_id" for relation "posthog_action_events" already exists
ERROR:  constraint "posthog_action_team_id_3a21e3a6_fk_posthog_team_id" for relation "posthog_action" already exists
ERROR:  constraint "posthog_actionstep_action_id_b50d75e6_fk_posthog_action_id" for relation "posthog_actionstep" already exists
ERROR:  constraint "posthog_annotation_created_by_id_1b9c9223_fk_posthog_user_id" for relation "posthog_annotation" already exists
ERROR:  constraint "posthog_annotation_dashboard_item_id_8bfb9aa9_fk_posthog_d" for relation "posthog_annotation" already exists
ERROR:  constraint "posthog_annotation_organization_id_f5c9d877_fk_posthog_o" for relation "posthog_annotation" already exists
ERROR:  constraint "posthog_annotation_team_id_95adb263_fk_posthog_team_id" for relation "posthog_annotation" already exists
ERROR:  constraint "posthog_cohort_created_by_id_e003077a_fk_posthog_user_id" for relation "posthog_cohort" already exists
ERROR:  constraint "posthog_cohort_team_id_8a849f3d_fk_posthog_team_id" for relation "posthog_cohort" already exists
ERROR:  constraint "posthog_cohortpeople_cohort_id_1f371733_fk_posthog_cohort_id" for relation "posthog_cohortpeople" already exists
ERROR:  constraint "posthog_cohortpeople_person_id_33da7d3f_fk_posthog_person_id" for relation "posthog_cohortpeople" already exists
ERROR:  constraint "posthog_dashboard_created_by_id_a528d2ca_fk_posthog_user_id" for relation "posthog_dashboard" already exists
ERROR:  constraint "posthog_dashboard_team_id_5c28ee1a_fk_posthog_team_id" for relation "posthog_dashboard" already exists
ERROR:  constraint "posthog_dashboardite_dashboard_id_59e2811f_fk_posthog_d" for relation "posthog_dashboarditem" already exists
ERROR:  constraint "posthog_dashboarditem_created_by_id_a3c81ce2_fk_posthog_user_id" for relation "posthog_dashboarditem" already exists
ERROR:  constraint "posthog_dashboarditem_team_id_a0e4bed0_fk_posthog_team_id" for relation "posthog_dashboarditem" already exists
ERROR:  constraint "posthog_element_event_id_bb6549a0_fk_posthog_event_id" for relation "posthog_element" already exists
ERROR:  constraint "posthog_element_group_id_09134876_fk_posthog_elementgroup_id" for relation "posthog_element" already exists
ERROR:  constraint "posthog_elementgroup_team_id_3ced0286_fk_posthog_team_id" for relation "posthog_elementgroup" already exists
ERROR:  constraint "posthog_eventdefinition_team_id_818ed0f2_fk_posthog_team_id" for relation "posthog_eventdefinition" already exists
ERROR:  constraint "posthog_featureflag_created_by_id_4571fe1a_fk_posthog_user_id" for relation "posthog_featureflag" already exists
ERROR:  constraint "posthog_featureflag_team_id_51e383b9_fk_posthog_team_id" for relation "posthog_featureflag" already exists
ERROR:  constraint "posthog_organization_created_by_id_16cbc2ef_fk_posthog_u" for relation "posthog_organizationinvite" already exists
ERROR:  constraint "posthog_organization_organization_id_15e2bab4_fk_posthog_o" for relation "posthog_organizationinvite" already exists
ERROR:  constraint "posthog_organization_organization_id_6f92360d_fk_posthog_o" for relation "posthog_organizationmembership" already exists
ERROR:  constraint "posthog_organization_user_id_aab9f3c7_fk_posthog_u" for relation "posthog_organizationmembership" already exists
ERROR:  constraint "posthog_personalapikey_team_id_813f490c_fk_posthog_team_id" for relation "posthog_personalapikey" already exists
ERROR:  constraint "posthog_personalapikey_user_id_730a29e7_fk_posthog_user_id" for relation "posthog_personalapikey" already exists
ERROR:  constraint "posthog_persondistin_person_id_5d655bba_fk_posthog_p" for relation "posthog_persondistinctid" already exists
ERROR:  constraint "posthog_persondistinctid_team_id_46330ec9_fk_posthog_team_id" for relation "posthog_persondistinctid" already exists
ERROR:  constraint "posthog_plugin_organization_id_d040b9a9_fk_posthog_o" for relation "posthog_plugin" already exists
ERROR:  constraint "posthog_pluginattach_plugin_config_id_cc94a1b9_fk_posthog_p" for relation "posthog_pluginattachment" already exists
ERROR:  constraint "posthog_pluginattachment_team_id_415eacc7_fk_posthog_team_id" for relation "posthog_pluginattachment" already exists
ERROR:  constraint "posthog_pluginconfig_plugin_id_d014ca1c_fk_posthog_plugin_id" for relation "posthog_pluginconfig" already exists
ERROR:  constraint "posthog_pluginconfig_team_id_71185766_fk_posthog_team_id" for relation "posthog_pluginconfig" already exists
ERROR:  constraint "posthog_pluginlogent_plugin_config_id_2fe30023_fk_posthog_p" for relation "posthog_pluginlogentry" already exists
ERROR:  constraint "posthog_pluginlogentry_plugin_id_42aaf74e_fk_posthog_plugin_id" for relation "posthog_pluginlogentry" already exists
ERROR:  constraint "posthog_pluginlogentry_team_id_5cab32b2_fk_posthog_team_id" for relation "posthog_pluginlogentry" already exists
ERROR:  constraint "posthog_pluginstorag_plugin_config_id_6744363a_fk_posthog_p" for relation "posthog_pluginstorage" already exists
ERROR:  constraint "posthog_propertydefinition_team_id_b7abe702_fk_posthog_team_id" for relation "posthog_propertydefinition" already exists
ERROR:  constraint "posthog_sessionrecor_team_id_5d0d59b9_fk_posthog_t" for relation "posthog_sessionrecordingviewed" already exists
ERROR:  constraint "posthog_sessionrecor_user_id_ef83047a_fk_posthog_u" for relation "posthog_sessionrecordingviewed" already exists
ERROR:  constraint "posthog_sessionsfilt_created_by_id_06700ae9_fk_posthog_u" for relation "posthog_sessionsfilter" already exists
ERROR:  constraint "posthog_sessionsfilter_team_id_59837f38_fk_posthog_team_id" for relation "posthog_sessionsfilter" already exists
ERROR:  constraint "posthog_team_organization_id_41bbc1d0_fk_posthog_o" for relation "posthog_team" already exists
ERROR:  constraint "posthog_team_users_team_id_2bd01272_fk_posthog_team_id" for relation "posthog_team_users" already exists
ERROR:  constraint "posthog_team_users_user_id_c7bafb47_fk_posthog_user_id" for relation "posthog_team_users" already exists
ERROR:  constraint "posthog_user_current_organization_e8527570_fk_posthog_o" for relation "posthog_user" already exists
ERROR:  constraint "posthog_user_current_team_id_52760528_fk_posthog_team_id" for relation "posthog_user" already exists
ERROR:  constraint "posthog_user_groups_group_id_75d77987_fk_auth_group_id" for relation "posthog_user_groups" already exists
ERROR:  constraint "posthog_user_groups_user_id_6b000a50_fk_posthog_user_id" for relation "posthog_user_groups" already exists
ERROR:  constraint "posthog_user_user_pe_permission_id_21270994_fk_auth_perm" for relation "posthog_user_user_permissions" already exists
ERROR:  constraint "posthog_user_user_pe_user_id_f94166b7_fk_posthog_u" for relation "posthog_user_user_permissions" already exists
ERROR:  constraint "rest_hooks_hook_user_id_94046fce_fk_posthog_user_id" for relation "rest_hooks_hook" already exists
ERROR:  constraint "social_auth_usersocialauth_user_id_17d28448_fk_posthog_user_id" for relation "social_auth_usersocialauth" already exists
ALTER TABLE
ALTER TABLE
ALTER TABLE
```

Reply:
seems this is related to my point from above. we need to wipe the db to restore using this method.

Reply:
Also hope you don't mind @leggetter but I've collapsed the error log